util/src/main/java/org/folio/metastorage/util/AsyncCodec.java

```diff
@@ -0,0 +1,58 @@
+package org.folio.metastorage.util;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+public class AsyncCodec {
+
+  /**
+   * Compress buffer with gzip.
+   * @param vertx context
+   * @param in input buffer
+   * @return future compressed buffer
+   */
+  public static Future<Buffer> compress(Vertx vertx, Buffer in) {
+    return vertx.executeBlocking(fut -> {
+      ByteArrayOutputStream baos = new ByteArrayOutputStream();
+      try (GZIPOutputStream gzos = new GZIPOutputStream(baos)) {
+        gzos.write(in.getBytes());
+        gzos.close();
```

Comment:
Not needed as that's what try does.

Reply:
Actually it doesn't, the promise completes before the try block and decompression fails with ZLIB exception. I have pulled a little bit of my remaining hair because of it.